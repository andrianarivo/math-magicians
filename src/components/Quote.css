h1 {
  font-size: 1.5rem;
  padding: 1em;
}

.info {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1em;
}

.error-image {
  width: 4em;
  height: 4em;
}
