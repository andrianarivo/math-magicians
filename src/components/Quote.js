import { useEffect, useState } from 'react';
import styles from '../styles/Quote.module.css';
import getQuote from '../logic/fetcher';
import loadingGif from '../assets/loading.gif';
import errorImage from '../assets/error.png';

function Quote() {
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(undefined);
  const [data, setData] = useState({});

  useEffect(() => {
    getQuote()
      .then((response) => {
        setData(response);
        setIsLoading(false);
      })
      .catch((reason) => {
        setError(reason);
        setIsLoading(false);
      });
  }, []);

  if (isLoading) {
    return (
      <div className={styles.info}>
        <img src={loadingGif} alt="loading dog" />
        <h3>Loading...</h3>
      </div>
    );
  }

  if (error) {
    return (
      <div className={styles.info}>
        <img className="error-image" src={errorImage} alt="error" />
        <h3>Sorry, an error occurred!</h3>
      </div>
    );
  }

  return <h3 className={styles.quote}>{`"${data[0].quote}"`}</h3>;
}

export default Quote;
