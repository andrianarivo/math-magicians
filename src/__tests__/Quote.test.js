import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

describe('testing Quote component', () => {
  test('render', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
