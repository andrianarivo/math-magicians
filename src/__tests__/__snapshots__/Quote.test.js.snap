// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Quote component render 1`] = `
<div
  className="info"
>
  <img
    alt="loading dog"
    src="loading.gif"
  />
  <h3>
    Loading...
  </h3>
</div>
`;
